### DISCUSSION TITLE
How to check whether Deno.child is still alive

### DISCUSSION THREAD

**Discussion by rizrmd** [denoland/deno#16622] [Q&A]
With deno unstable we can spawn a Deno.Child, but how to check whether it is still alive or not ?

**Comment by jtoppine** [comment]
You can await child.status, or use the promise like so:

```
child.status.then(() => {
  // process done, do stuff
});
```

**Reply by rizrmd** [reply]
I aware of this option, but I need to immediately tell whether child is already killed or not. My current workaround is like this 

```typescript
const isKilled = (child) => {
   return new Promise<boolean>((resolve) => {
       const timeout = setTimeout(() => {
           resolve(false);
       }, 1000);
       child.status.then(() => {
          clearTimeout(timeout)
          resolve(true);
       })
   }
}

```

This is not perfect, I need to wait at least X timeout (1s in above code) to check whether killed or not...

**Reply by rizrmd** [reply]
We have child.isKilled in nodejs as boolean, for comparation...

**Reply by rizrmd** [reply]
I think we can modify child itself to add isKilled property (defaults to false), and attach on "close" event handler to set it to true. Thanks @jtoppine !

**Comment by ghost** [comment]
Sounds like
```js
const is_child_alive = child =>
 Promise.race(
  child.status.then(_ => false),
  true
 );
```
Assuming it would be alive until its `.status` has resolved?

(I can't find the documentation!)

**Reply by rizrmd** [reply]
Oh wow, TIL... Promise.race
